### PR TITLE
FINERACT-2081: Upgrade Spring Batch to 5.1.2 version

### PR DESCRIPTION
**Pull request by alberto-art3ch**
## Description

In order to resolve some issues with remote workers, we are upgrading the Spring Batch version to 5.1.2

[Release v5.1.2 · spring-projects/spring-batch](https://github.com/spring-projects/spring-batch/releases/tag/v5.1.2)

[FINERACT-2081](https://issues.apache.org/jira/browse/FINERACT-2081)

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanCOBManagerConfiguration.java [hunk 1]
```diff
@@ -57,8 +57,6 @@ public class LoanCOBManagerConfiguration {
     @Autowired
     private PlatformTransactionManager transactionManager;
     @Autowired
-    private RemotePartitioningManagerStepBuilderFactory stepBuilderFactory;
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanCOBManagerConfiguration.java]: Why do you change this?

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Please kindly check my review!

File: fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanCOBManagerConfiguration.java (hunk 1 above)
Comment (alberto-art3ch) [reply fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanCOBManagerConfiguration.java]: This is not needed more

Comment (adamsaghy) [reply fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanCOBManagerConfiguration.java:60]: Injecting the factory is better than creating manually. Please undo this change!

Comment (alberto-art3ch) [reply fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanCOBManagerConfiguration.java]: The factory is deprecated, https://github.com/spring-projects/spring-batch/pull/4582/files

Comment (adamsaghy) [reply fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanCOBManagerConfiguration.java]: RemotePartitioningManagerStepBuilderFactory is not deprecated. Please dont remove it!

Comment (alberto-art3ch) [reply fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanCOBManagerConfiguration.java]: Change reverted

File: fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanCOBWorkerConfiguration.java [hunk 2]
```diff
@@ -86,7 +84,8 @@ public class LoanCOBWorkerConfiguration {
 
     @Bean(name = LoanCOBConstant.LOAN_COB_WORKER_STEP)
     public Step loanCOBWorkerStep() {
-        return stepBuilderFactory.get("Loan COB worker - Step").inputChannel(inboundRequests).flow(flow()).build();
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanCOBWorkerConfiguration.java]: Please revert this change...
